go/test/endtoend/vtgate/foreignkey/stress/fk_stress_test.go

```diff
@@ -0,0 +1,882 @@
+	updatesAttempts, updatesFailures, updatesNoops, updates int64
+	deletesAttempts, deletesFailures, deletesNoops, deletes int64
+}
+
+func (w *WriteMetrics) Clear() {
+	w.mu.Lock()
+	defer w.mu.Unlock()
+
+	w.inserts = 0
+	w.updates = 0
+	w.deletes = 0
+
+	w.insertsAttempts = 0
+	w.insertsFailures = 0
+	w.insertsNoops = 0
+
+	w.updatesAttempts = 0
+	w.updatesFailures = 0
+	w.updatesNoops = 0
+
+	w.deletesAttempts = 0
+	w.deletesFailures = 0
+	w.deletesNoops = 0
+}
+
+func (w *WriteMetrics) String() string {
+	return fmt.Sprintf(`WriteMetrics: inserts-deletes=%d, updates-deletes=%d,
+insertsAttempts=%d, insertsFailures=%d, insertsNoops=%d, inserts=%d,
+updatesAttempts=%d, updatesFailures=%d, updatesNoops=%d, updates=%d,
+deletesAttempts=%d, deletesFailures=%d, deletesNoops=%d, deletes=%d,
+`,
+		w.inserts-w.deletes, w.updates-w.deletes,
+		w.insertsAttempts, w.insertsFailures, w.insertsNoops, w.inserts,
+		w.updatesAttempts, w.updatesFailures, w.updatesNoops, w.updates,
+		w.deletesAttempts, w.deletesFailures, w.deletesNoops, w.deletes,
+	)
+}
+
+var (
+	clusterInstance *cluster.LocalProcessCluster
+	shards          []cluster.Shard
+	primary         *cluster.Vttablet
+	replica         *cluster.Vttablet
+	vtParams        mysql.ConnParams
+
+	hostname              = "localhost"
+	keyspaceName          = "ks"
+	cell                  = "zone1"
+	schemaChangeDirectory = ""
+	parentTableName       = "stress_parent"
+	childTableName        = "stress_child"
+	child2TableName       = "stress_child2"
+	grandchildTableName   = "stress_grandchild"
+	tableNames            = []string{parentTableName, childTableName, child2TableName, grandchildTableName}
+	reverseTableNames     []string
+
+	onDeleteActionsMap = map[sqlparser.ReferenceAction]string{
+		sqlparser.NoAction: "NO ACTION",
+		sqlparser.Cascade:  "CASCADE",
+		sqlparser.SetNull:  "SET NULL",
+	}
+	onDeleteActions  = []sqlparser.ReferenceAction{sqlparser.NoAction, sqlparser.Cascade, sqlparser.SetNull}
+	createStatements = []string{
+		`
+		CREATE TABLE stress_parent (
+			id bigint not null,
+			parent_id bigint,
+			rand_val varchar(32) null default '',
+			hint_col varchar(64) not null default '',
+			created_timestamp timestamp not null default current_timestamp,
+			updates int unsigned not null default 0,
+			PRIMARY KEY (id),
+			key parent_id_idx(parent_id),
+			key created_idx(created_timestamp),
+			key updates_idx(updates)
+		) ENGINE=InnoDB
+		`,
+		`
+		CREATE TABLE stress_child (
+			id bigint not null,
+			parent_id bigint,
+			rand_val varchar(32) null default '',
+			hint_col varchar(64) not null default '',
+			created_timestamp timestamp not null default current_timestamp,
+			updates int unsigned not null default 0,
+			PRIMARY KEY (id),
+			key parent_id_idx(parent_id),
+			key created_idx(created_timestamp),
+			key updates_idx(updates),
+			CONSTRAINT child_parent_fk FOREIGN KEY (parent_id) REFERENCES stress_parent (id) ON DELETE %s
+		) ENGINE=InnoDB
+		`,
```

Comment:
on update will be `RESTRICT` is that intentional?

Reply:
You're right. The entire test is only testing `ON DELETE`. I'll add `ON UPDATE`.

Reply:
The tests now validate all combinations of `ON DELETE` cross `ON UPDATE`. The tests validate `ON UPDATE CASCADE` and `ON UPDATE SET NULL` by issuing queries that modify the foreign key parent column (`id` in our case).